File: algorithms/java/src/test/java/leetcode/LC326PowerOfThree.java
package leetcode;

import org.junit.jupiter.api.Test;


import static org.junit.jupiter.api.Assertions.assertEquals;

/**
 * https://leetcode.com/problems/power-of-three/
 * #easy
 * <p/>
 * To do it without a loop resort to logarithms (but beware of accuracy)
 */
public class LC326PowerOfThree {
    public class Solution {
        public boolean isPowerOfThree(int n) {
            if (n < 1) return false;
            while (1 < n) {
                if (n % 3 != 0) {
                    return false;
                }
                n /= 3;
            }
            return true;
        }
    }

    @Test
    public void test_27() throws Exception {
        assertEquals(true, new Solution().isPowerOfThree(27));
    }

    @Test
    public void test_11() throws Exception {
        assertEquals(false, new Solution().isPowerOfThree(11));
    }

    @Test
    public void test_1() throws Exception {
        assertEquals(true, new Solution().isPowerOfThree(1));
    }

    @Test
    public void test_0() throws Exception {
        assertEquals(new Solution().isPowerOfThree(0), false);
    }

    @Test
    public void test_Minus3() throws Exception {
        assertEquals(new Solution().isPowerOfThree(-3), false);
    }
}

File: algorithms/code/leetcode/lc326_power_of_three/lc326_power_of_three.py
# https://leetcode.com/problems/power-of-three/
# To do it without a loop resort to logarithms (but beware of accuracy)

import unittest


class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        if n < 1:
            return False
        while n > 1:
            if n % 3 != 0:
                return False
            n //= 3
        return True


class TestCode(unittest.TestCase):
    def test_27(self) -> None:
        assert Solution().isPowerOfThree(27)

    def test_11(self) -> None:
        assert not Solution().isPowerOfThree(11)

    def test_1(self) -> None:
        assert Solution().isPowerOfThree(1)

    def test_0(self) -> None:
        assert not Solution().isPowerOfThree(0)

    def test_Minus3(self) -> None:
        assert not Solution().isPowerOfThree(-3)
